bindings/java/c/evmc-vm.c

```diff
@@ -112,6 +113,10 @@ JNIEXPORT jobject JNICALL Java_org_ethereum_evmc_EvmcVm_execute(JNIEnv* jenv,
     struct evmc_result* result =
         (struct evmc_result*)(*jenv)->GetDirectBufferAddress(jenv, jresult);
     assert(result != NULL);
+    size_t input_data_size;
+    const uint8_t* input_data = GetDirectBuffer(jenv, jinputdata, &input_data_size);
+    char* input_data_addr = (char*)msg + offsetof(struct evmc_message, input_data);
+    memcpy(input_data_addr, &input_data, sizeof(uint8_t*));
```

Comment:
Why does it need to use this convoluted approach of memory copying, as opposed to just `msg.input_data = GetDirectBuffer(jenv, jinputdata, &input_data_size);` ?

Reply:
Actually should be `msg.input_data = GetDirectBuffer(jenv, jinputdata, &msg.input_size);` as the size setting is missing right now.